openshift/console
# ODC-6779 - Hide sub-catalog(s) in the developer catalog or the entire dev catalog based on customization

JIRA story:
https://issues.redhat.com/browse/ODC-6779

Description:
This story is about the customization of developer catalog and sub-catalog based on customization. If all the available sub-catalogs(types) are disabled, complete developer catalog along with the entry points to the catalog will be disabled. 

Screenshots:
----When all the sub-catalogs are disabled - Add page----
<img width="1728" alt="Screenshot 2022-09-20 at 12 38 17 PM" src="https://user-images.githubusercontent.com/102503482/191235665-5da7407d-c424-488c-b438-d792693823d1.png">

----When all the sub-catalogs are disabled - Topology page----
<img width="1084" alt="Screenshot 2022-09-20 at 12 38 34 PM" src="https://user-images.githubusercontent.com/102503482/191235689-01e4ebf3-2840-4452-aeb3-0a43a3aa5fa1.png">

-----Quick start when complete developer catalog is disabled-----
<img width="923" alt="Screenshot 2022-09-20 at 12 38 52 PM" src="https://user-images.githubusercontent.com/102503482/191235677-010fc82d-dc7d-4c47-907f-146399f7e3ad.png">

------ All services - when all the sub-catalogs(types) are enabled----
<img width="566" alt="Screenshot 2022-09-20 at 2 18 35 PM" src="https://user-images.githubusercontent.com/102503482/191235607-203daf01-c5f7-4875-87cb-4afa03e52e49.png">

------ All services - when Templates and Helm charts sub-catalogs are disabled----
<img width="566" alt="Screenshot 2022-09-20 at 2 18 44 PM" src="https://user-images.githubusercontent.com/102503482/191235679-2b3b29bf-34dc-426a-8d4a-37b5286b109b.png">

-----Helm Page when helm chart sub-catalog is disabled-----
<img width="1728" alt="Screenshot 2022-09-20 at 12 38 01 PM" src="https://user-images.githubusercontent.com/102503482/191235687-4a3cf844-ea03-4884-bdc3-2750d89b6d99.png">


---- Operator hub description when operator backed service is disabled or complete developer catalog is disabled---
<img width="1436" alt="Screenshot 2022-09-20 at 12 39 37 PM" src="https://user-images.githubusercontent.com/102503482/191235660-99b7f770-d342-400e-92a6-b8d45e2b5924.png">

---- If Event source type is disabled, create event source in action menu is hidden----
<img width="1438" alt="Screenshot 2022-09-20 at 1 00 52 PM" src="https://user-images.githubusercontent.com/102503482/191235648-c9bb6bb1-0b88-4956-9475-edf346f54e13.png">

-----If user try to reach to catalog by entering URL and if that sub-catalog is disabled, then user will see Page not found--
<img width="1452" alt="Screenshot 2022-09-20 at 12 46 20 PM" src="https://user-images.githubusercontent.com/102503482/191235654-4a684993-d0b5-426a-9831-5ceab921ac0b.png">

Test cases:

<img width="719" alt="Screenshot 2022-09-20 at 3 26 05 PM" src="https://user-images.githubusercontent.com/102503482/191235663-179bd000-27bf-4043-92bf-2b70d9fe7497.png">

How to test:

If backend API PR's - https://github.com/openshift/api/pull/1256 & https://github.com/openshift/console-operator/pull/676 are not merged then to add the customization use below method,

Pass options as a string to bridge with -developer-catalog-disabled-types or pass a ConsoleConfig yaml with -config option.

Example:

Specify developer catalog types to bridge with ./bin/bridge -developer-catalog-types '{ "state": "Disabled" ,"disabled": ["HelmChart", "Devfile"] }'.

This indicated HelmChart and Devfile sub-catalogs are disabled. Similarly below are the test scenarios,

If the `state` is `Enabled` and `enabled` list is empty, all the types will be shown.
If the `state` id `Enabled` and `enabled` list is non-empty, only the `enabled` list items will be shown.
If the `state` is `Disabled` and `disabled` list is empty, all the types will be hidden.
If the `state` id `Disabled` and `disabled` list is non-empty, only the `disabled` list items will be hidden.

To disable the complete developer catalog then in `disabled` all the available types
should be listed or make the `disabled` list empty for `state: Disabled`. 


/kind feature
